test/javascripts/turbolinks_test.coffee

```diff
@@ -1,210 +1,946 @@
+            <link href='#{ASSET_FIXTURES['b.css']}' data-turbolinks-track="true"></link>
+            <link href='#{ASSET_FIXTURES['a.css']}' data-turbolinks-track="true"></link>
+            <title>Hi there!</title>
+          </head>
+          <body>
+            <div id="turbo-area" refresh="turbo-area"></div>
+          </body>
+        </html>
+      """
+    ],
+
+    fourLinksInHeadCBDA: [
+      200,
+      {'Content-Type':'text/html'},
+      """
+        <!doctype html>
+        <html>
+          <head>
+            <link href='#{ASSET_FIXTURES['c.css']}' data-turbolinks-track="true"></link>
+            <link href='#{ASSET_FIXTURES['b.css']}' data-turbolinks-track="true"></link>
+            <link href='#{ASSET_FIXTURES['d.css']}' data-turbolinks-track="true"></link>
+            <link href='#{ASSET_FIXTURES['a.css']}' data-turbolinks-track="true"></link>
+            <title>Hi there!</title>
+          </head>
+          <body>
+            <div id="turbo-area" refresh="turbo-area"></div>
+          </body>
+        </html>
+      """
+    ],
+
+    fourLinksInHeadDCBA: [
+      200,
+      {'Content-Type':'text/html'},
+      """
+        <!doctype html>
+        <html>
+          <head>
+            <link href='#{ASSET_FIXTURES['d.css']}' data-turbolinks-track="true"></link>
+            <link href='#{ASSET_FIXTURES['c.css']}' data-turbolinks-track="true"></link>
+            <link href='#{ASSET_FIXTURES['b.css']}' data-turbolinks-track="true"></link>
+            <link href='#{ASSET_FIXTURES['a.css']}' data-turbolinks-track="true"></link>
+            <title>Hi there!</title>
+          </head>
+          <body>
+            <div id="turbo-area" refresh="turbo-area"></div>
+          </body>
+        </html>
+      """
+    ],
+
+    inlineScriptInBody: [
+      200,
+      {'Content-Type':'text/html'},
+      """
+        <!doctype html>
+        <html>
+          <head>
+            <title>Hi</title>
+          </head>
+          <body>
+            <script id="turbo-area" refresh="turbo-area">globalStub()</script>
+          </body>
+        </html>
+        """
+    ],
+
+    inlineScriptInBodyTurbolinksEvalFalse: [
+      200,
+      {'Content-Type':'text/html'},
+      """
+        <!doctype html>
+        <html>
+          <head>
+            <title>Hi</title>
+          </head>
+          <body>
+            <script data-turbolinks-eval="false" id="turbo-area" refresh="turbo-area">globalStub()</script>
+          </body>
+        </html>
+      """
+    ],
+
+    responseWithRefreshAlways: [
+      200,
+      {'Content-Type':'text/html'},
+      """
+        <!doctype html>
+        <html>
+          <head>
+            <title>Hi</title>
+          </head>
+          <body>
+            <div id="div1" refresh="div1">
+              <div id="div2" refresh-always>Refresh-always</div>
+            </div>
+          </body>
+        </html>
+      """
+    ]
+  }
+
+  hasScript = (filename) ->
```

Comment:
If a helper is used exclusively by a single `describe` block, move the helper into the block's scope.  There's a strong case to be made for splitting this into 3 separate files; isolating the helpers makes that easier.